src/components/activity-page/managed-interactive/iframe-runtime.tsx

```diff
@@ -61,8 +54,7 @@ export const IframeRuntime: React.FC<IProps> =
       phone.post("initInteractive", {
         mode: report ? "report" : "runtime",
         authoredState,
-        // This is a trick not to depend on interactiveState.
-        interactiveState: interactiveStateRef.current
+        interactiveState: initialInteractiveState
```

Comment:
@pjanik I just wanted to point this modification out to you.
I guess in some way it is turning the iframe-runtime into an uncontrolled component. 

I'm somewhat paranoid with this component, because it seems like a re-usable piece, so whenever it diverges from other implementations I'd want to understand why.

I think the previous implementation (taken from question-interactives) was more like a controlled component. It did maintain its own reference to the interactiveState, but it was updating this reference on each render.

Reply:
I don't think there's a difference in the result. Previously we were disconnecting the `interactiveState` entirely from the parent component, such that if it changed in the props, we wouldn't reload the iframe. Indeed, if anything this was the "uncontrolled" version.

The change is now we have an `initialInteractiveState`, with a "promise" that we won't change it (hence the "initial" in the name), but if we were to change it we would reload the iframe. When it makes sense to update the initial state (e.g. we want to re-render the component because a dialog has changed) we will reload the component, which I think is what we would want. But currently since we don't change it it behaves as it did before.

Reply:
@scytacki, this is a bit tricky. The previous implementation wasn't a real controlled component. In fact it was uncontrolled one. If you think about the logic of this component, not much changed here. `(init)interactiveState` is sent to the iframe only during `initInteractive` callback in **both** implementations. We never sent it again to iframe when it was updated (that would a real controlled component), as we don't have Interactive API for that. So, in the prev implementation, if the parent component updated the `interactiveState` property, iframe-runtime would ignore that change (unless `authoredState` or `url` have been updated too). This was actually fine in question-interactives repo and could be okay here too (unless we want to support some external updates coming from Firestore).

I think I like this new implementation. It's more explicit and clear. `inititialInteractiveState` is really only the initial state at the moment. There is no way to update `interactiveState` by parent and this implementation makes it clear. It adds some work to the parent component, as it needs to ensure not to update `inititialInteractiveState` all the time, but it makes sense.

Once we add a clean way to update interactive state without reloading the iframe (probably new Interactive API call), we could just add a new property for this component called `interactiveState`. And parent component would have a choice whether it wants to use controller or uncontrolled version. If I remember correctly, that's how native inputs work in React - dev can choose whether to use `value` or `defaultValue` (which was called `initialValue` in some old React versions I think).

Reply:
Thanks for humoring me.  So then it sounds like we could copy this change back to the question-interactives if we want. No need to do it now, but it is good to know.